ThreeBotPackages/tfgrid/workloads/actors/workload_manager.py

```diff
@@ -372,7 +387,14 @@ def workloads_list(self, node_id, cursor, schema_out, user_session):
         ```
         """
         output = schema_out.new()
-        reservations = self._filter_reservations(node_id, ["deploy", "delete"], cursor)
+        reservations_subset = self._filter_reservations(node_id, ["deploy", "delete"], cursor)
+
+        reservations_actionable = self.reservation_actionable_model.find()
```

Comment:
This is very inefficient, loading the entire data in the reservation_actionable while we only need the ones for one specific node.

Reply:
Also in the Go implementation, the queue (resrvation_actionable in your) case has only "workloads" not full reservations. so I can immediately drop the workload from the queue once the node send a result for that workload (regardless it was delete/provision, or whatever) now you have to do some logic regard deleting to make sure all workloads in a reservation has been deleted before u remove that reservation from the queue.

also having the full reservation in the queue, means each time the node will poll it will get the same workload until all nodes report deletion for the workloads in that reservations.